### ISSUE TITLE
Chromebook Stream Stutter after 15 MBPS

### ISSUE THREAD

**Issue by Kylan-Straws** [moonlight-stream/moonlight-chrome#311]
Hello all,
Client: Samsung Chromebook Pro (Core m3, 4 GB RAM)
Server: Ryzen 5 1600 @ 3.8 Ghz, 8 GB DDR4 @ 2800, GTX 1060 6 GB, Intel 8165 Wireless Card. (I have a Killer 1535 card I can try if you guys believe it'll make a difference.)

I am using the Windows 10 mobile hotspot feature to connect my chromebook directly to the host PC, and latency is not noticeable. The host PC is under my bed, with me on the bed with a chromebook. The host pc has an antennae that sits on my nightstand, so no potential for any kind of interference that way.   

When playing at anywhere about 15 MBPS, I get stutters every 3-4 seconds which cause the game to be unplayable- this stutter does not occur on the host. This essentially puts me at 720p60 or a network pixelated 1080p60. I'd really like to get up to 30-50 mbps for a 1440p60 stream or super high quality 1060p60. 

Connecting directly to the computer via mobile hotspot works very well, and I'm assuming is better than going through a fast router. I do not have a fast router available at the moment, so it's a moot point for the time being.   

I'm wondering if I'm hitting a bottleneck of the software at 15 MBPS, and should maybe dual boot a linux distribution for more speed?

I'm also wondering if it's possible that the wireless cards in my system could be causing a bottleneck, or any other bottleneck you guys can think of.

**Comment by Jorys-Paulin** [comment]
Hey @Kylan-Straws,
- Could you try connecting your computers via an ethernet cable?
- I'm not familiar with the Win10 mobile hotspot, but since it's "mobile" I wouldn't be surprised if the bitrate is limited.
- Changing your wifi card would help, but only if the hardware can be used as its maximum

**Comment by Kylan-Straws** [comment]
@Jorys-Paulin 

- I'll pick up a crossover cable and let you know. Ideally I'd like everything to be wireless.
- It shouldn't be. It's the same technology as the "hostednetwork" wifi card, similar to what you needed Connectify for.
- I'll check with the other card and see if there's issues there.

**Comment by Jorys-Paulin** [comment]
@Kylan-Straws 
- Of course, but it's in case the connection isn't the problem.
- Well maybe it wasn't mean for high bitrate applications like this, but rather for web and mail viewing. The speed link between the two computers will give us the awnser.
- See if it's still the same or not.

**Comment by Kylan-Straws** [comment]
@Jorys-Paulin 
I was able to determine that the connection itself does support a super high bitrate - I was able to stream at even 100 MBPS to my phone with no stuttering at all using the Android version of Moonlight.
This laptop is much more powerful than my phone - and should be able to support this stream just fine. 

Any ideas?

**Comment by Jorys-Paulin** [comment]
So maybe your laptop's Wi-Fi card is the bottleneck then. What where the results using an Ethernet cable?

**Comment by Kylan-Straws** [comment]
It turns out it's not. I have been able to use the android version of moonlight on my chromebook with similar success to my phone.

I do have a solution for myself, but if you'd like to troubleshoot specifically for the chrome version, please let me know what information you need.

Otherwise we can close this, I'll just keep using the android version for now.

**Comment by Jorys-Paulin** [comment]
Wait, so you're running moonlight Android on your Chromebook, streaming from you host of to it, using a bridge connection by your phone?

**Comment by Kylan-Straws** [comment]
Nope, running Moonlight Android on my chromebook, connected to my Windows 10 PC directly through mobile hotspot, which uses the Wi-Fi Direct tech (apparently up to 250 mbps)

**Comment by Jorys-Paulin** [comment]
That means that chrome has to be the bottleneck, since the base C++ is the same for all versions (moonlight-common-c), but I'm not expert.

**Comment by Kylan-Straws** [comment]
Yes - I came to the same conclusion.
I'll try power washing the thing and kicking it out of dev mode and see if that improves it. Theoretically, I should be getting better performance on the Chrome version, right?

**Comment by Jorys-Paulin** [comment]
Meh, since we're running native code under browser code written in native code there should be some performance loss

**Comment by hifihedgehog** [comment]
I had the same problem but I found the cause and it was not packet loss as others have suggested, either. Because Moonlight is running within a Chrome session, other Chrome extensions are enabled and they can potentially interfere with the video streaming. It turns out if I have any of my ad-blocking and video downloading extensions enabled, Moonlight stutters incessantly. I simply disabled each extension one by one, and then the experience was buttery smooth and perfect from that point on. I strongly suggest to the developers to recommend that users disable other Chrome extensions when streaming or the streaming experience could likely suffer.
